src/Farmer/Arm/Databricks.fs

```diff
@@ -0,0 +1,65 @@
+[<AutoOpen>]
+module Farmer.Arm.Databricks
+
+open Farmer
+open Farmer.Databricks
+open Farmer.Arm.Network
+open System
+
+let workspaces = ResourceType ("Microsoft.Databricks/workspaces", "2018-04-01")
+
+type ByovConfig =
+    { Vnet: ResourceRef<ByovConfig>
+      PublicSubnet: ResourceName
+      PrivateSubnet: ResourceName }
+
+type EncryptionConfig =
+    { KeySource: string
+      KeyName: string
+      KeyVersion: string
+      KeyVault: ResourceName }
+
+type Databricks =
+    { Name: ResourceName
+      Location: Location
+      ManagedResourceGroupId: ResourceName
+      Sku: Sku
+      EnablePublicIp: FeatureFlag
+      PrepareEncryption: FeatureFlag
+      Encryption: EncryptionConfig option
+      ByovConfig: ByovConfig option
+      Tags: Map<string,string> }
+    interface IArmResource with
+        member this.ResourceId = workspaces.resourceId this.Name
+        member this.JsonModel =
+            {| workspaces.Create(this.Name, this.Location, tags = this.Tags) with
+                sku =
+                    {| name =
+                        match this.Sku with
+                        | StandardTier -> "standard"
+                        | PremiumTier -> "premium" |}
+                properties =
+                    {| managedResourceGroupId =
+                        let expr = sprintf "concat(subscription().id, '/resourceGroups/', '%s')" this.ManagedResourceGroupId.Value
+                        ArmExpression.create(expr).Eval()
+                       parameters = Map [
```

Comment:
I have moved this code over to a Map (think: key / value pair store) which allows us to programmatically create the fields. This removes the need for the "null" case and repeated pattern matching / option handling.

I also added a new helper, `Option.mapBoxed` which basically does `Option.map`, then boxes the result, and then does `Option.toObj`. This is a pattern I've seen in a few places and we could retrospectively apply this as a tech debt task.